doc/modules/cassandra/pages/architecture/storage-engine.adoc

```diff
@@ -102,6 +102,61 @@ If a node stops working, replaying the commit log restores writes to the memtabl
 
 Data in the commit log is purged after its corresponding data in the memtable is flushed to an SSTable on disk.
 
+=== Trie memtables
+
+An alternative memtable implementation based on tries, also called prefix trees, is provided alongside the legacy skip list solution. 
+The implementation is activated using the memtable API (https://cwiki.apache.org/confluence/display/CASSANDRA/CEP-11%3A+Pluggable+memtable+implementations[CEP-11] / https://issues.apache.org/jira/browse/CASSANDRA-17034[CASSANDRA-17034]).
+Trie memtables improve on the legacy solution in modification and lookup performance, as well as the size of the structure for a given amount of data.
+
+Trie memtables use a data structure called a trie to organize data. 
+This structure makes them very efficient at modifying and querying data, as well as more compact in memory. 
+These features result in higher write throughput, lower latency for accessing recently-written data, while fitting more of it in memory.
+
+Trie memtables have internal memory management mechanisms, which drastically reduce the amount of work needed for garbage collection, reducing GC-inflicted pauses and higher-percentile latencies.
+This improvement is crucial to {cassandra} as it reduces the impact of GC on the system and allows for more predictable performance.
+
+Trie memtables reduce write amplification, a common problem in database systems, by buffering and organizing writes until they fill up their allocated memory. 
```

Comment:
The first sentence applies to memtables generally; trie ones reduce it further by fitting more data.

Reply:
OK, I'll modify the sentence.

Reply:
You can simply drop "Trie" at the beginning of the sentence, or replace it with "Generally, ".